tiki/tiki-shopify
# Fix: Create a development enviroment

### Description

Creates a development environment to use Shopify CLI for testing local builds.

### How it was before
The Shopify CLI expects that both frontend and backend endpoints are served by a local node js server, built in web folder. Because of our serverless structure, it was not possible to run the code with Shopify CLI or test it in a Shopify development store without deploying the code to the respecting services, which delays the development cycle.

### What changed
The dev server is piped by Shopify CLI through a Cloudflare tunnel to localhost, where an express server is working in the web folder receiving all the requests from Shopify. 
The backend runs in a local wrangler deployment at localhost:8787 and the frontend is served as a static react app in localhost:3000. 
This way it is possible to embed the Admin UIs in Shopify and receive the api calls from the shop.

### How to test
TIKI app must be already installed in the store. This dev server does not handle the installation yet.
1. Run npm install in root, web, web/frontend and web/backend folders
2. In the root folder run npm run dev
3. Allow the reset of URLs when prompted by Shopify CLI
4. Navigate to Shopify admin and check the embedded URLs
